helm-chart/binderhub/templates/secret.yaml

```diff
@@ -3,18 +3,17 @@ apiVersion: v1
 metadata:
   name: binder-secret
 type: Opaque
-data:
-  {{- $values :=  dict "config" dict }}
-  {{- $cfg := .Values.config }}
-  binder.hub-token: {{ .Values.jupyterhub.hub.services.binder.apiToken | b64enc | quote }}
-  {{- /* every 'pick' here should be matched with a corresponding 'omit' in secret.yaml */ -}}
-  {{- if $cfg.GitHubRepoProvider }}
-  {{- $_ := set $values.config "GitHubRepoProvider" (pick $cfg.GitHubRepoProvider "client_id" "client_secret" "access_token") }}
-  {{- end }}
-  {{- if $cfg.GitLabRepoProvider }}
-  {{- $_ := set $values.config "GitLabRepoProvider" (pick $cfg.GitLabRepoProvider "access_token" "private_token") }}
-  {{- end }}
-  values.yaml: {{ $values | toYaml | b64enc | quote }}
+stringData:
+  {{- /* Make it easy to mount certain parts without traversing YAML */}}
+  binder.hub-token: {{ .Values.jupyterhub.hub.services.binder.apiToken | quote }}
+
+  {{- /* Stash away all Helm template values */}}
+  values.yaml: |
+    {{- .Values | toYaml | nindent 4 }}
```

Comment:
We've lost the logic [here](https://github.com/jupyterhub/binderhub/pull/1179/files#diff-1b900e3e30b9fbc60551b87d5f21bfbc14277cae86ba9441f391b2898a220958L11) which I think is important: not including quite all of values.yaml, since restarting the binder pods on every possible change to values is unnecessarily disruptive.

It can still be simpler since we don't have to do secret splitting, and it's okay to cast a slightly wider-than-strictly-necessary net in the name of simplicity, but I think restarting binder pods on every helm upgrade is too far, which this would do.

If we get that behavior back, then 👍 this is ready to go.

Reply:
Nice catch, thanks for the review @minrk! I updated this in dc3df7f